external/ilasm/ilasm.depproj

```diff
@@ -6,6 +6,22 @@
     <NugetRuntimeIdentifier>$(RuntimeOS)-x64</NugetRuntimeIdentifier>
     <OutputPath>$(ToolsDir)ilasm</OutputPath>
     <IsRuntimeAssembly>false</IsRuntimeAssembly>
+    <RidSpecificAssets>true</RidSpecificAssets>
   </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NETCore.Platforms">
+      <Version>2.0.0-$(CoreFxExpectedPrerelease)</Version>
+    </PackageReference>
+    <PackageReference Include="Microsoft.NETCore.Runtime.CoreCLR">
+      <Version>2.0.0-$(CoreClrExpectedPrerelease)</Version>
+    </PackageReference>
+    <PackageReference Include="Microsoft.NETCore.ILAsm">
+      <Version>2.0.0-$(CoreClrExpectedPrerelease)</Version>
+    </PackageReference>
+  </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
+  <!-- When zipped/unzipped in a package, ilasm loses its executable flag. Reset it. -->
+  <Target Name="MakeIlasmExecutable" AfterTargets="CopyFilesToOutputDirectory" Condition="'$(RunningOnUnix)' == 'true'">
```

Comment:
See this bullet point:

> * There are a bunch of new calls to chmod here and there. This is because .NET Core 2.0 (which we now restore all packages with, and run the build on) does not preserve executable flags in Unix files, by design. We need to mark our programs as executable before we try to use them.